### PR TITLE
Preserve Data Locally

### PR DESCRIPTION
**Pull request by danifromecuador**
## Completed requirements:

1. On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. `rentals.json`)

1. On startup of your app load all data from files. Remember to handle the case when one or more files are missing.

### CODE REVIEW

File: src/app.rb [hunk 1]
```diff
@@ -6,42 +6,178 @@
...
+    rentals_from_json
+  end
+
+  def exit_app
+    save_data_to_json
+    create_empty_json_file('./data/books.json') unless File.exist?('./data/books.json')
+    create_empty_json_file('./data/people.json') unless File.exist?('./data/people.json')
+    create_empty_json_file('./data/rentals.json') unless File.exist?('./data/rentals.json')
+    puts 'Exiting the application.'
+  end
```
Comment (1GORDON) [inline src/app.rb:67]: Great job team🤝

- [ ] As stated in the project requirements, on exit from the student's app, make sure the data is saved (all the data: books, people, and rentals) into JSON files. This is not the case for your project. I am getting this error whenever I exit the app.
json-error

- [ ] On startup of the app, load all data from files. Handle the case when one or more of the files are missing (the app should work and show no errors).

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @danifromecuador, @PabloBona, and @M-Anwar-Hussaini ✋

Good job so far!

## Highlights!
- Good UI ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/app.rb (hunk 1 above)
Comment (danmainah) [reply src/app.rb:67]: - [ ] @Team great work trying to implement the change requested, however am still getting the same error when I choose option 7 to exit the app, Kindly consider checking how you reference your JSON files.
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @danifromecuador , @PabloBona and @M-Anwar-Hussaini , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
✔️  Clean UI with all options working except the exit option.
✔️ Professional written README file with all sections required.


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/app.rb (hunk 1 above)
Comment (M-Anwar-Hussaini) [reply src/app.rb:67]: Hello @danmainah  and @1GORDON , Thanks for reviewing our code.
I think the problem is that you have not navigated to the `src` directory in your terminal while running the `main.rb` file.
Would you please first go to the `./src` folder by `cd src/` and then run the file again?

**Comment by M-Anwar-Hussaini** [comment]
> Hi @danifromecuador , @PabloBona and @M-Anwar-Hussaini ,
> 
> Great work on making the changes requested by a previous reviewer 👏 👏 👏 You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
> 
> ## Highlights
> ✔️ Clean UI with all options working except the exit option. ✔️ Professional written README file with all sections required.
> 
> # Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by M-Anwar-Hussaini** [comment]
Hello @danmainah  and @1GORDON , Thanks for reviewing our code.
I think the problem is that you have not navigated to the `src` directory in your terminal while running the `main.rb` file.
Would you please first go to the `./src` folder by `cd src/` and then run the file again?

It is working on our machine:

image

image

image

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @danifromecuador, @M-Anwar-Hussaini and @PabloBona,

✔️ Data are well preserved on app exit

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**